### DISCUSSION TITLE
[Question] What is the difference between Feature & FeatureGate

### DISCUSSION THREAD

**Discussion by chaomao** [jnunemaker/flipper#624] [General]
Hi, there
I try to understand Feature & FeatureGate, however I was still confused after read some source code, and no luck to find more detail on documentation.

would someone give me a hint?

**Comment by jnunemaker** [comment]
Are you asking about the active record models in the active record adapter or just "feature" vs "gate" in general?

**Reply by chaomao** [reply]
sorry for late response.

I am using active record adapter currently, but not sure this adapter may change business logic about `feature` and `gate`.

would you please explain both of them?

thanks in advance :)

**Reply by jnunemaker** [reply]
I do not recommend changing them. I'd consider them an internal representation. Nothing stopping you from copying them and the adapter into your own project and tweaking them however you see fit though.
